.github/actions/bumpVersion/bumpVersion.js

```diff
@@ -34,50 +37,42 @@ function updateNativeVersions(newVersion) {
 }
 
-// Use Github Actions' default environment variables to get repo information
-// https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables#default-environment-variables
-const [repoOwner, repoName] = process.env.GITHUB_REPOSITORY.split('/');
-
-// Determine current patch version
-const {version} = JSON.parse(fs.readFileSync('./package.json'));
-const currentPatchVersion = version.split('-')[0];
-console.log('Current patch version:', currentPatchVersion);
-
-let newVersion;
+const octokit = github.getOctokit(core.getInput('GITHUB_TOKEN', {required: true}));
+let semanticVersionLevel = core.getInput('SEMVER_LEVEL', {require: true});
 
-// Fetch tags
+// it actually would fall to build anyway, but I think it is better to make it explicitly
+if (!semanticVersionLevel || !_.find(versionUpdater.semanticVersionLevels, v => v === semanticVersionLevel)) {
+    console.log(
+        `Invalid input for 'SEMVER_LEVEL': ${semanticVersionLevel}`,
+        `Defaulting to: ${versionUpdater.semanticVersionLevels.build}`,
+    );
+    semanticVersionLevel = versionUpdater.semanticVersionLevels.build;
+}
 console.log('Fetching tags from github...');
-const octokit = github.getOctokit(core.getInput('GITHUB_TOKEN', {required: true}));
-return octokit.repos.listTags({
-    owner: repoOwner,
-    repo: repoName,
+octokit.repos.listTags({
+    owner: github.context.repo.owner,
+    repo: github.context.repo.repo,
 })
     .catch(githubError => core.setFailed(githubError))
     .then((githubResponse) => {
-        // Find the highest build version git tag
+        // Find the highest version git tag
         const tags = githubResponse.data.map(tag => tag.name);
-        console.log('Tags: ', tags);
-        const highestBuildNumber = Math.max(
-            ...(tags
-                .filter(tag => (tag.startsWith(currentPatchVersion)))
-                .map(tag => tag.split('-')[1])
-            ),
 
-            // Provide a default of 0 for minimum build version
-            0,
-        );
-        console.log('Highest build number from current patch version:', highestBuildNumber);
+        // tags come from latest to oldest
+        const highestVersion = tags[0];
+
+        console.log(`Highest version found: ${highestVersion}.`);
+
+        newVersion = versionUpdater.incrementVersion(highestVersion, semanticVersionLevel);
 
-        // Increment the build version, update the native and npm versions.
-        newVersion = `${currentPatchVersion}-${highestBuildNumber + 1}`;
         updateNativeVersions(newVersion);
         console.log(`Setting npm version for this PR to ${newVersion}`);
         return exec(`npm --no-git-tag-version version ${newVersion} -m "Update version to ${newVersion}"`);
     })
     .then(({stdout}) => {
         // NPM and native versions successfully updated, output new version
         console.log(stdout);
-        core.setOutput('newVersion', newVersion);
+        core.setOutput('VERSION', newVersion);
```

Comment:
```suggestion
        core.setOutput('NEW_VERSION', newVersion);
```

Reply:
I think we changed `VERSION` to `NEW_VERSION` in the action metadata file, so we need to change it here too.

Reply:
done!